esteid/middleware.py

```diff
@@ -69,9 +119,38 @@ def get_ocsp_responder_certificate_path(cls):
         return config.ocsp_responder_certificate_path()
 
     @classmethod
-    def verify_ocsp(cls, request):
-        issuer = request.META.get('HTTP_X_ISSUER', None)
-        certificate = request.META.get('HTTP_X_CLIENTCERT', None)
+    def verify_ocsp(cls, certificate, issuer):
+        """ Runs OCSP verification and returns error code - 0 means success
+        """
+
         return OCSPVerifier(certificate, issuer,
                             cls.get_ocsp_url(),
                             cls.get_ocsp_responder_certificate_path()).verify()
+
+
+class ApacheIdCardMiddleware(BaseIdCardMiddleware):
+    """ ID card middleware configured for Apache server
+
+    Requires standard SSL_CLIENT_S_DN, SSL_CLIENT_I_DN and SSL_CLIENT_CERT headers.
+    """
+
+    DN_HEADER_NAME = 'HTTP_SSL_CLIENT_S_DN'
+    ISSUER_HEADER_NAME = 'HTTP_SSL_CLIENT_I_DN'
+    CERTIFICATE_HEADER_NAME = 'HTTP_SSL_CLIENT_CERT'
+
+    def prepare_certificate(self, certificate):
+        return certificate.replace(' ', '\n').replace('\nCERTIFICATE-----', ' CERTIFICATE-----')
+
+    def parse_client_dn(self, dn):
+        return parse_rfc_dn(dn)
+
+
+class NginxIdCardMiddleware(BaseIdCardMiddleware):
+    DN_HEADER_NAME = 'HTTP_X_CLIENT'
+
+    # TODO: Set to a valid value once we add certificates support
+    CERTIFICATE_HEADER_NAME = None
```

Comment:
No cert support for nginx? Will this be added in the future? (add an issue if so)